src/core_plugins/kibana/public/management/sections/indices/create_index_pattern/create_index_pattern.js

```diff
@@ -207,39 +195,54 @@ uiModules.get('apps/management')
     return loadingCount > 0;
   };
 
-  this.refreshFieldList = () => {
-    const timeField = this.formValues.timeField;
+  this.refreshTimeFieldOptions = () => {
+    const prevOption = this.formValues.timeFieldOption;
+
     loadingCount += 1;
-    fetchFieldList().then(results => {
-      if (timeField) {
-        updateFieldListAndSetTimeField(results, timeField.name);
-      } else {
-        updateFieldList(results);
-      }
-    }).finally(() => {
-      loadingCount -= 1;
-    });
+    this.timeFieldOptions = [];
+    this.timeFieldOptionsError = null;
+    this.formValues.timeFieldOption = null;
+    getTimeFieldOptions()
+      .then(results => {
+        this.timeFieldOptions = results.options;
+        this.timeFieldOptionsError = results.error;
+        if (!this.timeFieldOptions) {
+          return;
+        }
+
+        const restoredOption = findTimeFieldOption(prevOption);
+        const defaultOption = pickDefaultTimeFieldOption();
+        this.formValues.timeFieldOption = restoredOption || defaultOption;
+      })
+      .catch(notify.error)
+      .finally(() => {
+        loadingCount -= 1;
+      });
   };
 
   this.createIndexPattern = () => {
-    const id = this.formValues.name;
-    let timeFieldName;
-    if ((this.formValues.timeField !== TIME_FILTER_FIELD_OPTIONS.NO_DATE_FIELD_DESIRED)
-      && (this.formValues.timeField !== TIME_FILTER_FIELD_OPTIONS.NO_DATE_FIELDS_IN_INDICES)) {
-      timeFieldName = this.formValues.timeField.name;
-    }
+    const {
+      name,
+      timeFieldOption,
+      nameIsPattern,
+      nameInterval,
+      expandable
+    } = this.formValues;
 
-    // Only event-time-based index patterns set an intervalName.
-    const intervalName =
-      this.formValues.nameIsPattern
-      ? this.formValues.nameInterval.name
-      : undefined;
+    const id = name;
```

Comment:
Yeah, I had it like that at one point, but found it easier to read when all of the values passed to `sendCreateIndexPatternRequest()` were defined in the same way (rather than using destructuring-rename for one and not others)